### DISCUSSION TITLE
`.pbtxt` is discouraged - rename all files to canonical `.txtpb`?

### DISCUSSION THREAD

**Discussion by InCogNiTo124** [triton-inference-server/server#6522] [Ideas]
Hello,

as per the [official protobuf documentation](https://protobuf.dev/reference/protobuf/textformat-spec/#text-format-files):

> .txtpb is the canonical text format file extension and should be preferred to the alternatives

Have you considered conforming all text proto files to this?

**Comment by the-david-oy** [comment]
Thanks for sharing this! @nnshah1, what are your thoughts on this?

**Comment by nnshah1** [comment]
@dyastremsky I believe it makes sense for us to add an item to the backlog to support the canonical name. We will have to support both .pbtxt and .txtpb for backwards compatibility.  eventually we could consider deprecating - but I see three stages:

1) support for .txtpb
2) update examples to use .txtpb
3) deprecate pbtxt [ timelne TBD]

@InCogNiTo124 is the priority to support .txtpb or to deprecate .pbtxt ?

**Reply by the-david-oy** [reply]
Thanks for clarifying! That timeline sounds good. Created DLIS-5780 to track this effort.

**Reply by InCogNiTo124** [reply]
IMO the priority should be to support `.txtpb` before deprecating `.pbtxt`, especially for new files
